src/jit/decomposelongs.cpp

```diff
@@ -920,15 +925,24 @@ GenTree* DecomposeLongs::DecomposeShift(LIR::Use& use)
 
     GenTree* tree   = use.Def();
     GenTree* gtLong = tree->gtGetOp1();
+    GenTree* oldShiftOp = tree->gtGetOp2();
+
     genTreeOps oper = tree->OperGet();
+    genTreeOps shiftOper = oldShiftOp->OperGet();
+
+    unsigned int count;
+    if (shiftOper == GT_CNS_INT)
+    {
+        count = oldShiftOp->gtIntCon.gtIconVal;
+    }
 
     assert((oper == GT_LSH) || (oper == GT_RSH) || (oper == GT_RSZ));
```

Comment:
can this all be within the shiftOper 'if', below? Did you put it here before shiftWidthOp got Remove()'d? But it already gets used after the remove.